opencensus-migration/src/main/java/OpenTelemetryContextManager.java

```diff
@@ -0,0 +1,39 @@
+import io.grpc.Context;
+import io.opencensus.trace.ContextManager;
+import io.opencensus.trace.Ctx;
+import io.opencensus.trace.Span;
+import io.opentelemetry.trace.TracingContextUtils;
+
+public class OpenTelemetryContextManager implements ContextManager {
+
+  private final SpanCache spanCache;
+
+  public OpenTelemetryContextManager() {
+    this.spanCache = SpanCache.getInstance();
+  }
+
+  @Override
+  public Ctx currentContext() {
+    return wrapContext(Context.current());
+  }
+
+  @Override
+  public Ctx withValue(Ctx ctx, Span span) {
+    OpenTelemetryCtx openTelemetryCtx = (OpenTelemetryCtx) ctx;
+    return wrapContext(
+        TracingContextUtils.withSpan(spanCache.toOtelSpan(span), unwrapContext(openTelemetryCtx)));
+  }
+
+  @Override
+  public Span getValue(Ctx ctx) {
+    return spanCache.fromOtelSpan(TracingContextUtils.getSpan(unwrapContext(ctx)));
```

Comment:
If you remember, we discussed that this may lead to a memory leak in one scenario: if the span was added to the context via OT, we won't have it in the cache. This means that `spanCache.fromOtelSpan` then will lead to creating an OC copy and adding it to the cache.

The problem is when OT span's `end()` is called we don't intercept it, and the item will never be removed from `spanCache`.

Two options to solve this:
1. Find a way to intercept OT span's `end()` and call `spanCache.remove(...)` from there.
2. Replace maps used in `SpanCache` with an "expiring cache" implementation, e.g. [Guava's cache]( https://guava.dev/releases/19.0/api/docs/com/google/common/cache/CacheBuilder.html) with `expireAfterWrite(10, TimeUnit.MINUTES)`  (let me know if you need help with configuring it)